enclave/ert.go

```diff
@@ -103,6 +107,75 @@ func VerifyRemoteReport(reportBytes []byte) (attestation.Report, error) {
 	}, verifyErr
 }
 
+// GetLocalReport gets a report signed by the enclave platform for use in local attestation.
+//
+// The report shall contain the data given by the reportData parameter. The report can only
+// hold a maximum of 64 byte reportData. Use a 64 byte hash value of your data as reportData
+// if your data exceeds this limit.
+//
+// The report can only be verified by the enclave identified by targetReport. So you must
+// first get a report from the target enclave. This report is allowed to be empty, i.e.,
+// obtained by `GetLocalReport(nil, nil)`.
+func GetLocalReport(reportData []byte, targetReport []byte) ([]byte, error) {
+	var report *C.uint8_t
+	var reportSize C.size_t
+
+	res, _, errno := syscall.Syscall6(
+		sysGetLocalReport,
+		getBytesPointer(reportData),
+		uintptr(len(reportData)),
+		getBytesPointer(targetReport),
+		uintptr(len(targetReport)),
+		uintptr(unsafe.Pointer(&report)),
+		uintptr(unsafe.Pointer(&reportSize)),
+	)
+	if err := oeError(errno, res); err != nil {
+		return nil, err
+	}
+
+	result := C.GoBytes(unsafe.Pointer(report), C.int(reportSize))
+	syscall.Syscall(sysFreeReport, uintptr(unsafe.Pointer(report)), 0, 0)
+	return result, nil
+}
+
+// VerifyLocalReport verifies the integrity of the local report and its signature.
+//
+// This function verifies that the report signature is valid. It
+// verifies that it is correctly signed by the enclave platform.
+//
+// The caller must verify the returned report's content.
+func VerifyLocalReport(reportBytes []byte) (attestation.Report, error) {
+	if len(reportBytes) <= 0 {
+		return attestation.Report{}, attestation.ErrEmptyReport
+	}
+
+	var report C.oe_report_t
+
+	res, _, errno := syscall.Syscall(
+		sysVerifyReport,
+		uintptr(unsafe.Pointer(&reportBytes[0])),
+		uintptr(len(reportBytes)),
+		uintptr(unsafe.Pointer(&report)),
+	)
+	if err := oeError(errno, res); err != nil {
+		return attestation.Report{}, err
+	}
+
+	if (report.identity.attributes & C.OE_REPORT_ATTRIBUTES_REMOTE) != 0 {
+		return attestation.Report{}, errors.New("OE_UNSUPPORTED")
```

Comment:
```suggestion
		return attestation.Report{}, errors.New("can't verify remote report: OE_UNSUPPORTED")
```
Returning why this report is not supported might be helpful. Not sure about the exact error message though.

Reply:
Agreed. I copied the code [from the past](https://github.com/edgelesssys/ego/blob/v0.1.0/enclave/enclave.go#L62-L76). Back then, I thought returning similar errors to OE would be a good idea. I don't think so anymore.